lib/nodeUtils.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function toJson(node) {
+    function deriveEndpointType(tags) {
+        const urlTypes = [];
+        Object.keys(tags).forEach((objectKey) => {
+            if (objectKey.indexOf('.url') !== -1 || objectKey.indexOf('.uri') !== -1) {
+                urlTypes.push(objectKey.substring(0, objectKey.length - 4).toUpperCase());
+            }
+        });
+
+        if (urlTypes.length > 0) {
+            return urlTypes[0];
+        }
+
+        return 'HTTP';
+    }
+
+    function deriveOperation(span) {
+        if (!span.getOperationName()) {
+            const httpMethod = span.getTags()['http.method'];
+            if (httpMethod) {
+                return httpMethod;
+            }
+        }
+        return span.getOperationName();
+    }
+
+    function deriveComponentType(tags) {
+        return tags.component;
+    }
+
+    function deriveUrl(tags) {
+        function parsePathFromURL(urlStr) {
+            let path = urlStr;
+            try {
+                path = require('url').parse(urlStr, true).pathname;
+            } catch (error) {}
+
+            return path;
+        }
+
+        let url = tags['http.url'];
+        if (url) {
+            return parsePathFromURL(url);
+        }
+
+        url = tags['http.uri'];
+        if (url) {
+            return parsePathFromURL(url);
+        }
+
+        url = tags['http.path'];
+        if (url) {
+            return url;
+        }
+
+        return null;
+    }
+
+    function tagsToProperties(tags) {
+        if (!tags || Object.keys(tags).length === 0) {
+            return undefined;
+        }
+
+        const properties = [];
+        Object.keys(tags).forEach((objectKey) => {
+            properties.push({
+                name: objectKey,
+                value: tags[objectKey],
+                type: 'Text',
```

Comment:
This is ok for now, but opentracing captures properties of numeric type as well, so if provided as a number, then we would want to record it as such. (Not handled in a special way currently in the backend, but hopefully soon :)